### PR TITLE
Bump golang.org/x/net to v0.12.0

### PR DESCRIPTION
**Pull request by fangn2**
*Issue #, if available:*

*Description of changes:*
Bump golang.org/x/net used in `examples/cmd/snapshotting` from `v0.0.0-20220728211354-c7608f3a8462` to `v0.12.0` 
https://pkg.go.dev/golang.org/x/net@v0.12.0?tab=versions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: examples/cmd/snapshotting/go.mod [hunk 1]
```diff
@@ -36,7 +36,8 @@ require (
 	github.com/vishvananda/netlink v1.2.1-beta.2 // indirect
 	github.com/vishvananda/netns v0.0.0-20211101163701-50045581ed74 // indirect
 	go.mongodb.org/mongo-driver v1.10.0 // indirect
-	golang.org/x/net v0.0.0-20220728211354-c7608f3a8462 // indirect
+	golang.org/x/net v0.10.0 // indirect
```
Comment (swagatbora90) [inline examples/cmd/snapshotting/go.mod:39]: The PR mentions golang.org/x/net latest version as `v0.12.0` but this is still `v0.10.0`?

Comment (fangn2) [reply examples/cmd/snapshotting/go.mod:39]: I was planning to upgrade to latest which is v0.12.0, then told by CI that some dependencies introduced by v0.12.0 cannot be built with current go version. 
So downgraded to v0.10.0.  
Yet noticed another issue which requires go 1.17. 
I will upgrade that first.

Comment (fangn2) [reply examples/cmd/snapshotting/go.mod]: I have PR #498 upgrading go to 1.17 merged.
And switched back to upgrade target version to v.012.0 which is latest.

**Comment by fangn2** [comment]
Buildkite CI is failing on building [Firecracker from tip of Firecracker's main branch](https://buildkite.com/firecracker-microvm/firecracker-go-sdk/builds/3210#0189410e-2759-4cb1-80ab-5c642ec9187c) which is not related to this change.
The [CI](https://buildkite.com/firecracker-microvm/firecracker-go-sdk/builds?branch=main) has been failing for the same reason in the last 9 consecutive nightly builds.

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ jobs:
   build:
     strategy:
       matrix:
-        go: ['1.14', '1.15', '1.16', '1.17', '1.18', '1.19']
+        go: ['1.17', '1.18', '1.19', '1.20']
```
Comment (fangn2) [inline .github/workflows/go.yml:14]: The x/sys version upgrade in the PR requires go version 1.17.  Removing old versions from the build matrix.

File: .buildkite/pipeline.yml [hunk 3]
```diff
@@ -136,7 +145,10 @@ steps:
       distro: "${BUILDKITE_AGENT_META_DATA_DISTRO}"
       hostname: "${BUILDKITE_AGENT_META_DATA_HOSTNAME}"
 
-  - wait
+  # This makes sure next step will run even if previous steps failed
+  # to make sure we can reach the final step for resource cleanup 
+  - wait: ~
+    continue_on_failure: true
```
Comment (fangn2) [inline .buildkite/pipeline.yml:151]: This is needed to guarantee we always able to reach to final stage for cleanup. 

Previously, if we have failures in early stages, the build will stop. Multiple failures without resource cleanup will eat up the storage resulting [No space left error](https://buildkite.com/firecracker-microvm/firecracker-go-sdk/builds/3219#0189429f-e175-48cf-8bdc-10e0775ebf1a).

**Review by turan18 — APPROVED** [review]

File: .buildkite/pipeline.yml (hunk 3 above)
Comment (swagatbora90) [reply .buildkite/pipeline.yml:151]: I would suggest decoupling the pipeline changes from the dependency upgrade. Split it into two commits or two PRs

Comment (fangn2) [reply .buildkite/pipeline.yml:151]: I splitted into two  commits. 
PTAL @swagatbora90

**Review by swagatbora90 — APPROVED** [review]

**Pull request merged**